io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -601,10 +602,26 @@ impl LvsLvol for Lvol {
     fn is_snapshot(&self) -> bool {
         unsafe { spdk_blob_is_snapshot(self.blob_checked()) }
     }
-    /// Returns a boolean indicating if the lvol is a clone.
-    fn is_clone(&self) -> bool {
-        unsafe { spdk_blob_is_clone(self.blob_checked()) }
+
+    /// Lvol is considered as clone if its sourceuuid attribute is a valid
+    /// snapshot. if it is clone, return the snapshot lvol.
+    fn is_snapshot_clone(&self) -> (bool, Option<Lvol>) {
+        if let Some(source_uuid) =
+            Lvol::get_blob_xattr(self, CloneXattrs::SourceUuid.name())
+        {
+            let snap_lvol =
+                match UntypedBdev::lookup_by_uuid_str(source_uuid.as_str()) {
+                    Some(bdev) => match Lvol::try_from(bdev) {
+                        Ok(l) => l,
+                        _ => return (false, None),
+                    },
+                    None => return (false, None),
```

Comment:
Doesn't seem entirely correct here, self is indeed a clone, but the Snapshot Lvol is missing?

Reply:
let snap_lvol =
                match UntypedBdev::lookup_by_uuid_str(source_uuid.as_str()) {
                    Some(bdev) => match Lvol::try_from(bdev) {
                        Ok(l) => l,
                        **_ => return None,**
                    },
                    **None => return None,**
                };
Above 2 `return None` condition is to handle the failure scenario of `lookup_by_uuid_str` and `try_from` function. Am I missing any point here ?